modules/gapi/src/executor/gexecutor.cpp

```diff
@@ -155,15 +155,18 @@ void cv::gimpl::GExecutor::run(cv::gimpl::GRuntimeArgs &&args)
 #if !defined(GAPI_STANDALONE)
             // Building as part of OpenCV - follow OpenCV behavior
             // if output buffer is not enough to hold the result, reallocate it
-            auto& out_mat   = *get<cv::Mat*>(args.outObjs.at(index));
-            createMat(desc, out_mat);
-#else
+            if (cv::util::holds_alternative<cv::Mat*>(args.outObjs.at(index)))
+            {
+                auto& out_mat   = *get<cv::Mat*>(args.outObjs.at(index));
+                createMat(desc, out_mat);
+                continue;
```

Comment:
this `continue`-under-`ifdef` is mind-breaking.

Please avoid constructions like this because it is VERY unclear how you first get `cv::Mat` as `args.outObjs.at(index)` and then, a few lines below, get `own::Mat*` from exactly the same object.

Reply:
Fixed